impl/observable.ts

```diff
@@ -0,0 +1,496 @@
+/**
+MIT License

```

Comment:
Hmm, should we perhaps make this a separate LICENSE file in the top-level of the repo?

Reply:
I think that's fine. Mostly I like licensing things because of the liability.